irohad/ametsuchi/command_executor.hpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_COMMANDEXECUTOR_HPP
+#define IROHA_COMMANDEXECUTOR_HPP
+
+#include <dao/command.hpp>
+
+namespace iroha {
+
+  namespace ametsuchi {
+    /**
+     * Applies command to the world state view
+     */
+    class CommandExecutor {
+      /**
+       * Executes a command in a temporary state
+       * @see TemporaryWsv, MutableStorage
+       * @param command Command to execute
+       * @return True if the command is successfully executed, false otherwise
+       */
+      virtual bool execute(dao::Command command) = 0;
```

Comment:
Does Ametsuchi have executor?
I heard Ametsuchi is only for database and doesn't have logic and stateful validator.